test-flakiness

```diff
@@ -79,16 +88,22 @@ begin
       result.each do |test_name, stats|
         test_stats_rollup[test_name] ||= {
           started: [],
-          failed: []
+          failed: [],
+          time: [],
         }
         test_stats_rollup[test_name][:started].concat(stats[:started])
         test_stats_rollup[test_name][:failed].concat(stats[:failed])
+        test_stats_rollup[test_name][:time].concat(stats[:time])
       end
     end
   ) do |build|
     # Skip this build if it's not on the requested branch
     next unless build.branch? options[:branch_name]
 
+    # Skip this build if it failed. While it may have failed due to flakes, we
+    # don't want to count legitmate failures as flakes
+    next if build.failed?
```

Comment:
Just to clarify, we're counting error results as flaky, but not failures (assertion failures)?

Reply:
Assertion failures are still counted as flakes as long as the test passed on rerun. I'm only skipping if the build was red overall.

The underestimates overall flakiness, especially for very flaky tests, but hopefully it also reduces noise by not counting UI tests failures on builds that were legitimately red.